test/test_optimality_conditions.jl

```diff
@@ -21,6 +21,25 @@ check_optimality(f::IndBallL1, x, gamma, y) = begin
     return all(sign_is_correct) && check_optimality(IndSimplex(f.r), abs.(x), gamma, abs.(y))
 end
 
+check_optimality(f::NormTV, x, gamma, y) = begin
+    N = length(x)
+    # compute dual solution
+    u = zeros(N+1)
+    u[1] = 0
+    for k in 2:N+1
+        u[k] = x[k-1]-y[k-1]+u[k-1]
+    end
+
+    # check whether all duals in interval
+    c1 = all(abs.(u) .<= gamma*f.lambda + 1e-10)
+    # check whether last equals 0 (first by construction)
+    c2 = isapprox(u[end], 0, atol=1e-12)
+    # check whether equal +- gamma*lambda
+    h = sign.(y[1:end-1] - y[2:end])
+    c3 = all(isapprox.( u[2:end-1] .* abs.(h) , h *f.lambda*gamma))
+    return c1 && c2 && c3
+end
+
 test_cases = [
```

Comment:
You also need to add some test cases to this list (analogously to the other ones here, just using the total variation function), so that we can look deeper into the failing checks